substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrRecorderThread.java

```diff
@@ -130,8 +130,10 @@ private void persistBuffers(JfrChunkWriter chunkWriter) {
                 boolean shouldNotify = persistBuffer(chunkWriter, buffer);
                 if (shouldNotify) {
                     Object chunkRotationMonitor = JavaVersionUtil.JAVA_SPEC >= 20
-                                    ? Target_jdk_jfr_internal_JVM.CHUNK_ROTATION_MONITOR
-                                    : Target_jdk_jfr_internal_JVM.FILE_DELTA_CHANGE;
+                    // CHUNK_ROTATION_MONITOR is available in Java 17 as well starting with 17.0.7
+                                    || (JavaVersionUtil.JAVA_SPEC == 17 && (JavaVersionUtil.JAVA_INTERIM > 0 || JavaVersionUtil.JAVA_UPDATE >= 7))
+                                                    ? Target_jdk_jfr_internal_JVM.CHUNK_ROTATION_MONITOR
+                                                    : Target_jdk_jfr_internal_JVM.FILE_DELTA_CHANGE;
```

Comment:
I am not sure it will work. The return value needs to be known at compile time so that the compiler drops the missing field access as dead code, otherwise it will try to access the wrong field (at compile time) and fail.

Reply:
Done with the help of a static final field :)